include/TTipHit.h

```diff
@@ -27,18 +28,42 @@ class TTipHit : public TGRSIDetectorHit {
     Double_t slow_amplitude;
     Double_t gamma_amplitude;
    
-    Double_t fit_time;
+	Int_t	 tip_channel;
+
+    Double_t    time_fit;
+    Double_t    sig2noise;
 
   public:
     /////////////////////////    /////////////////////////////////////
-    inline void SetFilterPattern(const int &x)   { filter   = x; }   //! 
+    inline void SetFilterPattern(const int &x)    { filter   = x; }   //! 
     inline void SetPID(Double_t x)                { fPID = x;     }   //!
+	inline void SetTipChannel(const int x)		  { tip_channel = x; } //!
 
     inline Int_t    GetFiterPatter()           { return filter;   }  //!
     inline Double_t GetPID()                   { return fPID;      }  //!
+	inline Double_t GetFitTime()			   { return time_fit;	} //!
+	inline Double_t GetSignalToNoise()		   { return sig2noise;	} //!
+	inline Int_t	GetTipChannel()			   { return tip_channel; } //!
 
     bool   InFilter(Int_t);                                         //!
 
+    void SetVariables(TFragment &frag) { SetCfd(frag.GetCfd());
+                                         SetCharge(frag.GetCharge());
+                                         SetTimeStamp(frag.GetTimeStamp()); }
+
+	void SetUpNumbering(TChannel &chan) { MNEMONIC mnemonic;
+										  TChannel *channel = GetChannel();
+										  if(!channel){
+										    Error("SetDetector","No TChannel exists for address %u",GetAddress());
+										    return;
+										  }
+										  ClearMNEMONIC(&mnemonic);
+										  ParseMNEMONIC(channel->GetChannelName(),&mnemonic); 
+										  Int_t tmp = (int16_t)atoi(mnemonic.arraysubposition.c_str()); 
+										  SetTipChannel((Int_t)(10*mnemonic.arrayposition + tmp)); }
+
```

Comment:
I see what the tip_channel is now. You should take a look at how we did Griffin. We have a crystal number in TGriffinHit (you can call this subdetector), and we use the detector in TGRSIDetectorHIt. You can then make the "crystal" transient and build it on the fly like the detector number. If you want to get the "human understandable" channel number, we use a function called GetArrayNumber, which returns fdetector*4+fcrystal. I think you can do the same here with fdetector*10+fsubdetector.